src/classes/BDI_HelpTextCMT.cls

```diff
@@ -0,0 +1,249 @@
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Rollups
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Wrapper classes for the Help Text CustomMetadataType objects used to create the
+* default Custom Metadata records
+*/
+public class BDI_HelpTextCMT {
+
+    /**
+     * @description Enum of the supported MetadataType objects (without the __mdt suffix)
+     */
+    public enum MetadataObject {
+        Data_Import_Field_Mapping,
+        Data_Import_Field_Mapping_Set,
+        Data_Import_Object_Mapping,
+        Data_Import_Object_Mapping_set
+    }
+
+    /** @description Default value for the Protected_x property when inserting new Custom Metadata Type records */
+    private static final Boolean IS_PROTECTED = false;
+
+    /**
+     * Wrapper class for Data_Import_Field_Mapping__mdt used to create
+     * the default CMT data as well as for use in the Custom UI.
+     */
+    public class DataImportFieldMapping {
+        public String recordName;
+        public String label;
+        public String dataImportObjectName;
+        public String dataImportFieldMappingSetRecordName;
+        public String dataImportFieldMappingSetLabel;
+        public String sourceFieldAPIName;
+        public String targetFieldAPIName;
+        public String targetObjectMappingRecordName;
+        public String targetObjectMappingLabel;
+        public String isRequired;
+        public Boolean isDeleted;
+
+        public DataImportFieldMapping(String label) {
+            this.label = label.trim();
+        }
+
+        public DataImportFieldMapping(String sourceFieldAPIName, MappingDetail detail) {
+            this.label = detail.dataImportFieldLabel;
+            this.sourceFieldAPIName = sourceFieldAPIName;
+            this.targetFieldAPIName = detail.targetFieldAPIName;
+            this.dataImportObjectName = detail.dataImportObjectName;
+            this.recordName = generateRecordName(this.label, true, null, this.dataImportObjectName);
+            this.isRequired = 'No';
+            this.isDeleted = false;
+        }
+
+        public Metadata.CustomMetadata getMetadataRecord() {
+            Metadata.CustomMetadata md = new Metadata.CustomMetadata();
+
+            String mdTypeName = UTIL_Namespace.StrTokenNSPrefix(BDI_HelpTextCMT.MetadataObject.Data_Import_Field_Mapping.Name() + '__mdt');
+
+            // Setup custom metadata to be created or updated in the subscriber org.
+            Metadata.CustomMetadata customMd = new Metadata.CustomMetadata();
```

Comment:
Do we need both the 'md' and the 'customMd' instances? What's the thinking there?